pkg/executables/config/kind.yaml

```diff
@@ -25,6 +25,11 @@ containerdConfigPatches:
 {{- else }}
         ca_file = "/etc/containerd/certs.d/{{.RegistryMirrorEndpoint}}/ca.crt"
 {{- end }}
+{{- if .RegistryAuth }}
+      [plugins."io.containerd.grpc.v1.cri".registry.configs."{{.RegistryMirrorEndpoint}}".auth]
+        username = "{{.RegistryUsername}}"
+        password = "{{.RegistryPassword}}"
```

Comment:
How are we handling passwords containing quotes? Or usernames for that matter (albeit that's odd).

It looks like we could do with some more testing around this.

Reply:
I can do more testing around this and follow up in a different pr if any changes are necessary